frontend/src/app/sessions/user-sessions-wrapper/create-sessions/create-persistent-session/create-persistent-session.component.ts

```diff
@@ -116,4 +117,10 @@ export class CreatePersistentSessionComponent implements OnInit {
         }
       });
   }
+
+  get toolsWithWorkspaceEnabled(): Observable<Tool[] | undefined> {
+    return this.toolService.tools$.pipe(
+      map((tools) => tools?.filter((tool) => tool.config.workspaces.enabled)),
+    );
+  }
```

Comment:
I would suggest moving this to the tool service itself and using a similar method as we already have in the user session service, which would look like this (if you applied some renaming, the example needs to be adjusted):

```ts
public readonly toolsWithWorkspaceEnabled$ = this.tools$.pipe(
  filter(Boolean),
  map((tools) => tools.filter((tool) => tool.config.workspaces.enabled),
) 
```

Theoretically, we could also add an `isToolWorkspaceEnabled` function, but that is only useful if we plan to use it somewhere else in the future, so I assume we can skip that here.

Reply:
Services are always loaded globally and global variables should only be used where absolutely necessary (e.g., to prevent code duplication). It makes the toolService larger & harder to understand.

When we start to create Observables to receive each different value of tools, this will end up in many unnested Observables. 

Since it's only used in this component and has a clear scope in the component, I wouldn't move it to the service.